modules/xfem/src/efa/EFAFragment2D.C

```diff
@@ -334,345 +334,86 @@ EFAFragment2D::split()
-      EFANode * node3 = _boundary_edges[1]->getNode(1);
-      EFANode * node4 = _boundary_edges[3]->getEmbeddedNode(0);
-      EFANode * node5 = _boundary_edges[2]->getNode(1);
-
-      new_frag1->addEdge(new EFAEdge(node1, node2));
-      new_frag1->addEdge(new EFAEdge(node2, node3));
-      new_frag1->addEdge(new EFAEdge(node3, node4));
-      new_frag1->addEdge(new EFAEdge(node4, node1));
-
-      new_frag2->addEdge(new EFAEdge(node3, node5));
-      new_frag2->addEdge(new EFAEdge(node5, node4));
-      new_frag2->addEdge(new EFAEdge(node4, node3));
-
-      new_fragments.push_back(new_frag1);
-      new_fragments.push_back(new_frag2);
-    }
-    else if (cut_edges[0] == 3 && cut_nodes[0] == 2) // case 5
-    {
-      EFAFragment2D * new_frag1 = new EFAFragment2D(_host_elem, false, NULL);
-      EFAFragment2D * new_frag2 = new EFAFragment2D(_host_elem, false, NULL);
-      EFANode * node1 = _boundary_edges[0]->getNode(0);
-      EFANode * node2 = _boundary_edges[0]->getNode(1);
-      EFANode * node3 = _boundary_edges[1]->getNode(1);
-      EFANode * node4 = _boundary_edges[3]->getEmbeddedNode(0);
-      EFANode * node5 = _boundary_edges[2]->getNode(1);
-
-      new_frag1->addEdge(new EFAEdge(node1, node2));
-      new_frag1->addEdge(new EFAEdge(node2, node3));
-      new_frag1->addEdge(new EFAEdge(node3, node4));
-      new_frag1->addEdge(new EFAEdge(node4, node1));
-
-      new_frag2->addEdge(new EFAEdge(node3, node5));
-      new_frag2->addEdge(new EFAEdge(node5, node4));
-      new_frag2->addEdge(new EFAEdge(node4, node3));
+    fragment_nodes[frag_number].push_back(_boundary_edges[iedge]->getNode(0));
 
-      new_fragments.push_back(new_frag1);
-      new_fragments.push_back(new_frag2);
-    }
-    else if (cut_edges[0] == 3 && cut_nodes[0] == 2) // case 6
+    if (_boundary_edges[iedge]->getNode(0)->category() ==
+        EFANode::N_CATEGORY_EMBEDDED_PERMANENT) // if current node has been cut change fragment
     {
-      EFAFragment2D * new_frag1 = new EFAFragment2D(_host_elem, false, NULL);
-      EFAFragment2D * new_frag2 = new EFAFragment2D(_host_elem, false, NULL);
-      EFANode * node1 = _boundary_edges[0]->getNode(0);
-      EFANode * node2 = _boundary_edges[0]->getEmbeddedNode(0);
-      EFANode * node3 = _boundary_edges[0]->getNode(1);
-      EFANode * node4 = _boundary_edges[1]->getNode(1);
-      EFANode * node5 = _boundary_edges[2]->getNode(1);
-
-      new_frag1->addEdge(new EFAEdge(node1, node2));
-      new_frag1->addEdge(new EFAEdge(node2, node4));
-      new_frag1->addEdge(new EFAEdge(node4, node5));
-      new_frag1->addEdge(new EFAEdge(node5, node1));
-
-      new_frag2->addEdge(new EFAEdge(node2, node3));
-      new_frag2->addEdge(new EFAEdge(node3, node4));
-      new_frag2->addEdge(new EFAEdge(node4, node2));
-
-      new_fragments.push_back(new_frag1);
-      new_fragments.push_back(new_frag2);
+      ++node_cut_count;
+      if (frag_number == 0)
+        frag_number = 1;
+      else
+        frag_number = 0;
```

Comment:
these could be replaced by 

```
frag_number = 1 - frag_number;
```

Reply:
That wouldn't work if `frag_number` is 0, would it?